File: src-all/C++/S1314-matrix-block-sum.cpp
/*
Given a m x n matrix mat and an integer k, return a matrix answer where each answer[i][j] is the sum of all elements mat[r][c] for:

	i - k <= r <= i + k,
	j - k <= c <= j + k, and
	(r, c) is a valid position in the matrix.

 
Example 1:
Input: mat = [[1,2,3],[4,5,6],[7,8,9]], k = 1
Output: [[12,21,16],[27,45,33],[24,39,28]]

Example 2:
Input: mat = [[1,2,3],[4,5,6],[7,8,9]], k = 2
Output: [[45,45,45],[45,45,45],[45,45,45]]

 
Constraints:
	m == mat.length
	n == mat[i].length
	1 <= m, n, k <= 100
	1 <= mat[i][j] <= 100


*/
#include <cassert>
#include <iostream>

class Solution {
public:
    vector<vector<int>> matrixBlockSum(vector<vector<int>>& mat, int k) {
        
    }
};

int main() {
  Solution sol;
  assert(sol.matrixBlockSum(0) == 0);
  std::cout << "Pass test cases!" << std::endl;
}

File: src-all/Rust/S1314-matrix-block-sum.rs
/*
Given a m * n matrix mat and an integer K, return a matrix answer where each answer[i][j] is the sum of all elements mat[r][c] for i - K <= r <= i + K, j - K <= c <= j + K, and (r, c) is a valid position in the matrix.
 
Example 1:
Input: mat = [[1,2,3],[4,5,6],[7,8,9]], K = 1
Output: [[12,21,16],[27,45,33],[24,39,28]]

Example 2:
Input: mat = [[1,2,3],[4,5,6],[7,8,9]], K = 2
Output: [[45,45,45],[45,45,45],[45,45,45]]

 
Constraints:
	m == mat.length
	n == mat[i].length
	1 <= m, n, K <= 100
	1 <= mat[i][j] <= 100

*/
pub struct Solution {}
impl Solution {
    pub fn matrix_block_sum(mat: Vec<Vec<i32>>, k: i32) -> Vec<Vec<i32>> {
        
    }
}

fn main() {
  assert_eq!(0, Solution::matrix_block_sum(0));
  println!("Pass test cases!");
}
